app/helpers/moments_helper.rb

```diff
@@ -36,11 +61,88 @@ def moments_stats
     end
     result + '</div>'
   end
+
+  def setup_moment_or_strategy(element)
+    if element.class.name == 'Moment'
+      return {
+        url_helper: moment_path(element),
+        link: edit_moment_path(element),
+        moods: element.mood.map { |item| Mood.find(item).name },
+        story_type: t('moments.singular')
+      }
+    end
+    {
+      url_helper: strategy_path(element),
+      link: edit_strategy_path(element),
+      moods: nil,
+      story_type: t('strategies.singular')
+    }
+  end
+
+  def user_actions(element, variables_to_present_type)
+    if element.user_id == current_user&.id
+      return {
+        viewers: get_viewer_list(element.viewers, nil),
+        edit:
+          {
+            link: variables_to_present_type[:link],
+            name: t('common.actions.edit')
+          },
+        delete:
+          {
+            name: t('common.actions.delete'),
+            link: variables_to_present_type[:url_helper],
+            dataMethod: 'delete',
+            dataConfirm: t('common.actions.confirm')
+          }
+      }
+    end
+    {
+      viewers: nil,
+      edit: nil,
+      delete: nil
+    }
+  end
+
+  def present_moment_or_strategy(element)
+    avatar = ProfilePicture.normalize_url(User.find(element.user_id).avatar.url)
+    variables_to_present_type = setup_moment_or_strategy(element)
+    user_actions = user_actions(element, variables_to_present_type)
+    {
+      name: element.name,
+      link: variables_to_present_type[:url_helper],
+      date: TimeAgo.created_or_edited(element),
+      actions: {
+        edit: user_actions[:edit],
+        delete: user_actions[:delete],
+        viewers: user_actions[:viewers]
+      },
+      draft: !element.published? ? t('draft') : nil,
+      categories: categories(element),
+      moods: variables_to_present_type[:moods],
+      storyBy: {
+        author: link_to(
+          User.find(element.user_id).name,
+          profile_index_path(uid: User.find_by(id: element.user_id).uid)
+        ),
+        avatar: avatar
+      },
+      storyType: variables_to_present_type[:story_type]
+    }
+  end
   # rubocop:enable MethodLength
 
   private
 
   # rubocop:disable MethodLength
+
+  def categories(element)
```

Comment:
This method can be written as

`Category.where(id: element.category).map(&:name)`

or

`Category.find(element.category).map(&:name)`


I would however recommend having it in the models